### ISSUE TITLE
Has anyone gotten `typos.config` to work on Neovim?

### ISSUE THREAD

**Issue by mawkler** [tekumara/typos-vscode#31]
I'm trying to explicitly set the typos config using `typos.config`. However, it doesn't seem to get picked up. I've tried specifying the config value in all kinds of ways, but none of them seems to work:

```lua
require('lspconfig').typo_lsp.setup({
  init_options = {
    config = '/custom/path/.typos.toml',
    config = {'/custom/path/.typos.toml'},
  },
  config = {'/custom/path/.typos.toml'},
  config = '/custom/path/.typos.toml',
  settings = {
    typos = {
      config = {'/custom/path/.typos.toml'},
      config = '/custom/path/.typos.toml',
    }
  }
})
```

What am I doing wrong?

**Comment by tekumara** [comment]
`config` is received via initializationOptions so it goes in [init_options](https://neovim.io/doc/user/lsp.html#lsp-config:~:text=triggers%20the%20command.-,init_options,-Values%20to%20pass), eg:
```lua
require('lspconfig').typos_lsp.setup({
    init_options = {
      config = '/custom/path/.typos.toml',
    }
})
```

However, it looks like there's an edge case! When Neovim doesn't initialise typo_lsp with `workspaceFolders`  (ie: `workspaceFolders = vim.NIL`) the config file is ignored 😱 This is something that typos-lsp should handle, so I'll look into making a fix.

PS: It looks like `workspaceFolders` is [derived from root_dir](https://neovim.io/doc/user/lsp.html#lsp-config:~:text=file%20is%20found.-,workspace_folders,-list%20of%20%7B%20uri) by default (ie: when not explicitly set). However, if `workspaceFolders` is not explicitly set, and root_dir is nil (because it doesn't contain [any of these files](https://github.com/neovim/nvim-lspconfig/blob/8917d2c830e04bf944a699b8c41f097621283828/lua/lspconfig/server_configurations/typos_lsp.lua#L7)) then we end up with `rootPath = vim.NIL, rootUri = vim.NIL, workspaceFolders = vim.NIL` and therefore an ignored config file 😞

**Comment by mawkler** [comment]
Interesting, thanks for looking into it!

**Comment by tekumara** [comment]
Hopefully this is now fixed in 0.1.12, let me know how it goes 🤞

**Comment by mawkler** [comment]
@tekumara Is the v0.1.12 release supposed to show up under [releases](https://github.com/tekumara/typos-vscode/releases)? It still seems to show v0.1.11 as the latest version 🤔

**Comment by tekumara** [comment]
Unfortunately there's a [32-bit windows build error](https://github.com/tekumara/typos-vscode/actions/runs/7773800970/job/21197912529) I haven't managed to fix yet blocking the release 😞

**Comment by tekumara** [comment]
Fixed and v0.1.12 has been released, let me know how you go 🤞

**Comment by mawkler** [comment]
@tekumara Yes, my issue has been fixed! Thank you very much!
